go/processor/floria/run_context.go

```diff
@@ -61,36 +61,31 @@ func (r runContext) executeCall(kind tosca.CallKind, parameters tosca.CallParame
 		r.static = true
 	}
 
-	if r.blockParameters.Revision >= tosca.R09_Berlin &&
-		!isPrecompiled(recipient, r.blockParameters.Revision) &&
-		!isStateContract(recipient) &&
-		!r.AccountExists(recipient) &&
-		parameters.Value.Cmp(tosca.Value{}) == 0 {
+	isStateContract := isStateContract(parameters.CodeAddress)
+	isPrecompiled := isPrecompiled(parameters.CodeAddress, r.blockParameters.Revision)
+
+	if kind == tosca.Call && r.blockParameters.Revision >= tosca.R09_Berlin && !isPrecompiled &&
+		!isStateContract && !r.AccountExists(parameters.Recipient) && parameters.Value.Cmp(tosca.Value{}) == 0 {
```

Comment:
nit:
```suggestion
	if kind == tosca.Call &&
	 r.blockParameters.Revision >= tosca.R09_Berlin &&
	 !isPrecompiled &&
 	 !isStateContract &&
 	 !r.AccountExists(parameters.Recipient) &&
 	 parameters.Value.Cmp(tosca.Value{}) == 0 {
```
this is a crowded condition. I think separating it like this could help reading it. wdyt?